src/wasm/wasm-s-parser.cpp

```diff
@@ -2452,17 +2468,22 @@ void SExpressionWasmBuilder::parseExport(Element& s) {
   ex->name = s[1]->str();
   if (s[2]->isList()) {
     auto& inner = *s[2];
-    ex->value = inner[1]->str();
+
     if (elementStartsWith(inner, FUNC)) {
+      ex->value = getFunctionName(*inner[1]);
```

Comment:
Why these changes to non-table exports?

Reply:
In all cases, it doesn't handle numerical index to the module element. Check the following:

```
a.wast:
(module
  (import "env" "f1" (func $f1 (param) (result)))
  (export "f1" (func 0))
)

$ bin/wasm-opt a.wast --print

Output:

[wasm-validator error in module] unexpected false: module function exports must be found, on 
0
(module
 (type $none_=>_none (func))
 (import "env" "f1" (func $f1))
 (export "f1" (func $0))
)

Fatal: error validating input
```

Reply:
I can make this a separate PR if that's what you meant.

Reply:
No, that makes sense. Thanks for the explanation!